src/main/scala/org/hammerlab/guacamole/readsets/ReadsRDD.scala

```diff
@@ -0,0 +1,26 @@
+package org.hammerlab.guacamole.readsets
+
+import org.apache.spark.rdd.RDD
+import org.hammerlab.guacamole.reads.{MappedRead, PairedRead, Read}
+
+/**
+ * A thin wrapper around an RDD[Read], with helpers to filter to mapped and paired-mapped reads.
+ */
+case class ReadsRDD(reads: RDD[Read], sourceFile: String) {
+  lazy val mappedReads =
+    reads.flatMap {
+      case r: MappedRead                   => Some(r)
+      case PairedRead(r: MappedRead, _, _) => Some(r)
+      case _                               => None
+    }
+
+  lazy val mappedPairedReads: RDD[PairedRead[MappedRead]] =
+    reads.flatMap {
+      case rp: PairedRead[_] if rp.isMapped => Some(rp.asInstanceOf[PairedRead[MappedRead]])
+      case _                                => None
+    }
+}
+
+object ReadsRDD {
+  def apply(pair: (RDD[Read], String)): ReadsRDD = ReadsRDD(pair._1, pair._2)
```

Comment:
This constructor describes how to make a `ReadsRDD` from a `(RDD[Read], String)` tuple. I believe the class should own the logic for constructing instances of itself from variously-shaped inputs, rather than lysing such boilerplate out into call-sites 😄 .

FWIW, this is currently used in [one place](https://github.com/hammerlab/guacamole/pull/432/files#diff-91485b818b79f0449c217f8f2ad7580aR61), and I prefer letting it say

```
.map(ReadsRDD(_))
```

instead of

```
.map(t => ReadsRDD(t._1, t._2))
```